custom_components/control4/climate.py

```diff
@@ -0,0 +1,434 @@
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement used by the platform."""
+        if self._extra_state_attributes["SCALE"] == "F":
+            return UnitOfTemperature.FAHRENHEIT
+        else:
+            return UnitOfTemperature.CELSIUS
+
+    @property
+    def supported_features(self) -> ClimateEntityFeature:
+        """Flag supported features."""
+        features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE
+            | ClimateEntityFeature.FAN_MODE
+            | ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+        )
+        if "Emergency" in self._extra_state_attributes["HVAC_MODES_LIST"]:
+            features = features | ClimateEntityFeature.AUX_HEAT
+        #    _LOGGER.debug( "Emergency Found: %s",
+        #        self._extra_state_attributes["HVAC_MODES_LIST"],
+        # )
+        return features
+
+    async def async_set_hvac_mode(self, hvac_mode) -> None:
+        """Set the hvac mode."""
+        c4_climate = self.create_api_object()
+
+        #
+        _LOGGER.debug(
+            "set hvac mode: %s",
+            hvac_mode,
+        )
+        if hvac_mode == HVACMode.HEAT:
+            if self.aux_mode_active:
+                _LOGGER.debug(
+                    "set hvac mode with aux: %s",
+                    hvac_mode,
+                )
+                await c4_climate.setHvacMode(CONTROL4_HVAC_MODE_AUX_HEAT)
+            else:
+                await c4_climate.setHvacMode(CONTROL4_HVAC_MODE_HEAT)
+        else:
+            if hvac_mode in CONTROL4_HVAC_MODES:
+                await c4_climate.setHvacMode(CONTROL4_HVAC_MODES[hvac_mode])
+            else:
+                _LOGGER.exception(
+                    "Request for unsupported hvac mode received:: %s",
+                    hvac_mode,
+                )
+
+    async def async_set_fan_mode(self, fan_mode) -> None:
+        """Set new target fan mode."""
+        c4_climate = self.create_api_object()
+        if fan_mode in CONTROL4_FAN_MODES:
+            await c4_climate.setFanMode(CONTROL4_FAN_MODES[fan_mode])
+        else:
+            _LOGGER.exception(
+                "Request for unsupported hvac mode received:: %s",
+                hvac_mode,
+            )
+
+    #    async def async_set_humidity(self, humidity) -> None:
+    #        """Set new target humidity."""
+    #        c4_climate = self.create_api_object()
+    #        await c4_climate.setHumidity(humidity)
+
+    async def async_set_temperature(self, **kwargs) -> None:
+        """Set new target temperature."""
+        c4_climate = self.create_api_object()
+        low_temp = kwargs.get(ATTR_TARGET_TEMP_LOW)
+        high_temp = kwargs.get(ATTR_TARGET_TEMP_HIGH)
+        temp = kwargs.get(ATTR_TEMPERATURE)
+        try:
+            if self.hvac_mode == HVACMode.HEAT_COOL:
+                if low_temp and high_temp:
+                    if high_temp - low_temp < MIN_TEMP_RANGE:
```

Comment:
Why do we need to enforce a MIN_TEMP_RANGE in Home Assistant? Does Control4 not enforce this on the device side?